src/test/java/org/opensearch/knn/index/KNNSettingsTests.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index;
+
+import lombok.SneakyThrows;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.network.NetworkModule;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.ByteSizeValue;
+import org.opensearch.env.Environment;
+import org.opensearch.knn.KNNTestCase;
+import org.opensearch.knn.plugin.KNNPlugin;
+import org.opensearch.node.MockNode;
+import org.opensearch.node.Node;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.InternalTestCluster;
+import org.opensearch.test.MockHttpTransport;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.test.NodeRoles.dataNode;
+
+public class KNNSettingsTests extends KNNTestCase {
+
+    @SneakyThrows
+    public void testGetSettingValueFromConfig() {
```

Comment:
I assume that would fail without changes in rebuild cache? do we need to include other settings as well?

Reply:
Not sure I understand what you mean by "that would fail without changes in rebuild cache". Could you elaborate?

Reply:
> I mean is this test, if it exists before your PR, fail without you code change?

Yes, this was the failure before the fix:
```
  2> REPRODUCE WITH: ./gradlew ':test' --tests "org.opensearch.knn.index.KNNSettingsTests.testGetSettingValueFromConfig" -Dtests.seed=2B1E0A6CFDBD324B -Dtests.security.manager=false -Dtests.locale=fr -Dtests.timezone=America/Paramaribo -Druntime.java=11
  2> java.lang.AssertionError: expected:<8126464> but was:<13>
        at __randomizedtesting.SeedInfo.seed([2B1E0A6CFDBD324B:4423BA039518D35]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
```

I think my actual and expected are mixed up, but Ill go ahead and fix this.

> And second question is about settings, we're checking here only for KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_LIMIT, but we have other settings like KNNSettings.KNN_CACHE_ITEM_EXPIRY_ENABLED and KNNSettings.KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES, do we need to have similar test for them as well?

I think it is probably okay to test just one. We group all of the dynamic cache settings here: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/KNNSettings.java#L223 and then dont handle them uniquely from there, so I think we are okay.